### ISSUE TITLE
Suport for tar.gz

### ISSUE THREAD

**Issue by eskhool** [TeamWin/Team-Win-Recovery-Project#473]
Does TWRP support tar.gz files for flashing? Is adding support feasible? 
Reason: We can support delta upgrades using the tar and gz formats over ANY regular http/server/file based distribution which is trickier/impossible to do with .zip format

**Comment by Dees-Troy** [comment]
No, we don't support tar.gz file flashing. Something like this could be added, but I don't see a real need for it. Is this something you need for a business?

**Comment by eskhool** [comment]
The use case for us is delta upgrades. Right now there is no ready solution to do a delta zip (atleast not without running a rsync server which is computationally not feasible and not scalable). There are open source solutions which allow a delta sync for a .tar.gz file over a normal HTTP server which for nightly/weekly ROM downloads can reduce download size to 15-30% (in my testing). but since the best recoveries like TWRP don't support .tar.gz out of the box the zip has to be converted at both ends which is not feasible.

**Comment by dosomder** [comment]
Since the maximum size of a flashable zip is 2147483647bytes (2GB), I would also be interested in this in the hope of allowing flashable files to be up to 4GB large.

Is there any update for this?

**Comment by AmyClaraRose** [comment]
what would you need a 4gb flash file for most roms are only 250mb and system partions only go as high as 2gb

**Comment by dosomder** [comment]
Current flagships of Sony have a 5GB system partition. Compressed it's around 2.2 GB which exceeds the maximum size of a flashable zip.
